debezium-core/src/main/java/io/debezium/converters/CloudEventsConverter.java

```diff
@@ -0,0 +1,549 @@
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import io.debezium.annotation.VisibleForTesting;
+import io.debezium.config.Configuration;
+import io.debezium.config.Instantiator;
+import io.debezium.data.Envelope;
+import io.debezium.util.SchemaNameAdjuster;
+
+/**
+ * Implementation of Converter that express schemas and objects with CloudEvents specification. The serialization
+ * format can be Json or Avro.
+ * <p>
+ * The serialization format of CloudEvents is configured with
+ * {@link CloudEventsConverterConfig#CLOUDEVENTS_SERIALIZER_TYPE_CONFIG cloudevents.serializer.type} option.
+ * <p>
+ * The serialization format of the data attribute in CloudEvents is configured with
+ * {@link CloudEventsConverterConfig#CLOUDEVENTS_DATA_SERIALIZER_TYPE_CONFIG cloudevents.data.serializer.type} option.
+ * <p>
+ * If the serialization format of the data attribute in CloudEvents is JSON, it can be configured whether to enable
+ * schemas with {@link CloudEventsConverterConfig#CLOUDEVENTS_JSON_SCHEMAS_ENABLE_CONFIG json.schemas.enable} option.
+ * <p>
+ * If the the serialization format is Avro, the URL of schema registries for CloudEvents and its data attribute can be
+ * configured with {@link CloudEventsConverterConfig#CLOUDEVENTS_SCHEMA_REGISTRY_URL_CONFIG schema.registry.url}.
+ * <p>
+ * There are two modes for transferring CloudEvents as Kafka messages: structured and binary. In the structured content
+ * mode, event metadata attributes and event data are placed into the Kafka message value section using an event format.
+ * In the binary content mode, the value of the event data is placed into the Kafka message's value section as-is,
+ * with the content-type header value declaring its media type; all other event attributes are mapped to the Kafka
+ * message's header section.
+ * <p>
+ * Since Kafka converters has not support headers yet, right now CloudEvents converter use structured mode as the
+ * default.
+ */
+public class CloudEventsConverter implements Converter {
+
+    private static final String EXTENSION_NAME_PREFIX = "iodebezium";
+
+    /**
+     * Instantiated reflectively to avoid hard dependency to Avro converter.
+     */
+    private static final String AVRO_CONVERTER_CLASS = "io.confluent.connect.avro.AvroConverter";
+    private static final String SCHEMA_REGISTRY_URL_CONFIG = "schema.registry.url";
+
+    /**
+     * Suffix appended to schema names of data schema in case of Avro/Avro, to keep
+     * data schema and CE schema apart from each other
+     */
+    private static final String DATA_SCHEMA_SUFFIX = "-data";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CloudEventsConverter.class);
+    private static Method CONVERT_TO_CONNECT_METHOD;
+
+    static {
+        try {
+            CONVERT_TO_CONNECT_METHOD = JsonConverter.class.getDeclaredMethod("convertToConnect", Schema.class, JsonNode.class);
```

Comment:
This is pretty hacky, but I think it's ok to resolve it in a follow-up (DBZ-1701); the reason being that this isn't invoked by `fromConnectData()` (which is the main usage of this converter) but only during deserialization, for which this converter not really is intended to be used atm.